src/Sarif/Visitors/SarifCurrentToVersionOneVisitor.cs

```diff
@@ -655,6 +655,10 @@ private Stream GetContentStream(Uri uri, out Encoding encoding)
                             {
                                 failureReason = $"File '{uri.LocalPath}' could not be accessed: {ex.ToString()}";
                             }
+                            catch (UnauthorizedAccessException ex)
```

Comment:
>UnauthorizedAccessException [](start = 35, length = 27)

IOException, SecurityException and UnauthorizedAccessException are the exceptions that need to be handled for System.IO API that actually touch the file system (some API, such as the path construction API might not actually need to interop with the file system). Someday we could write a little action/delegate helper that wraps a handler of this kind and returns any exception that's been raised to DRY out this pattern. Might be a nice small thing for Harleen.

Reply:
Consider DRYing this out with a C# 6.0 exception filter:
```
catch (Exception ex) when (
    ex is IOException ||
    ex is SecurityException ||
    ex is UnauthorizedAccessException)
{
    failureReason = $"File '{uri.LocalPath}' could not be accessed: {ex}"
}
```

(Also note that you don't need to call `ToString()`.)

---
In reply to: [255589393](https://github.com/Microsoft/sarif-sdk/pull/1276#discussion_r255589393) [](ancestors = 255589393)

Reply:
This code isn't the focus of the change, added UnauthorizedAccessException here for completeness. We'll clean this up later, sure.

---
In reply to: [255709280](https://github.com/Microsoft/sarif-sdk/pull/1276#discussion_r255709280) [](ancestors = 255709280,255589393)